var/lib/delphix-platform/os-migration

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+MIGRATE_CONFIG_SCRIPT="/opt/delphix/migration/migrate_config.py"
+MIGRATE_CONFIG_LOG="/var/delphix/migration/log"
+
+function die() {
+	echo "$(basename "$0"): $*" >&2
+	exit 1
+}
+
+function perform_migration() {
+	zpool import -f domain0 || die "Failed to import domain0"
+	# domain0/mds is owned by delphix:staff on Illumos and staff has GID 10.
+	# To minimize the changes in case we need to perform rollback, we add
+	# the appropriate group to the postgres user instead of changing the
+	# permissions of domain0/mds.
+	usermod -a -G 10 postgres ||
+		die "Failed adding group 10 to postgres user"
+	echo "$(date): Running migrate_config post-upgrade" \
+		>>"$MIGRATE_CONFIG_LOG"
```

Comment:
will `/var/delphix/migration` already exist? if it doesn't, what will happen?

Reply:
`/var/delphix/migration` will already exist; if it doesn't then migration would fail.